src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -73,29 +73,34 @@ public class MavenMojoProjectParser {
 
     private final Log logger;
     private final Path baseDir;
-    private final MavenProject mavenProject;
-    private final List<Marker> projectProvenance;
     private final boolean pomCacheEnabled;
     @Nullable
     private final String pomCacheDirectory;
     private final boolean skipMavenParsing;
+
+    private final BuildTool buildTool;
+
     private final Collection<String> exclusions;
     private final int sizeThresholdMb;
     private final MavenSession mavenSession;
     private final SettingsDecrypter settingsDecrypter;
 
     @SuppressWarnings("BooleanParameter")
-    public MavenMojoProjectParser(Log logger, Path baseDir, boolean pomCacheEnabled, @Nullable String pomCacheDirectory, MavenProject mavenProject, RuntimeInformation runtime, boolean skipMavenParsing, Collection<String> exclusions, int thresholdMb, MavenSession session, SettingsDecrypter settingsDecrypter) {
+    public MavenMojoProjectParser(Log logger, Path baseDir, boolean pomCacheEnabled, @Nullable String pomCacheDirectory, RuntimeInformation runtime, boolean skipMavenParsing, Collection<String> exclusions, int thresholdMb, MavenSession session, SettingsDecrypter settingsDecrypter) {
         this.logger = logger;
         this.baseDir = baseDir;
-        this.mavenProject = mavenProject;
         this.pomCacheEnabled = pomCacheEnabled;
         this.pomCacheDirectory = pomCacheDirectory;
         this.skipMavenParsing = skipMavenParsing;
+        this.buildTool = new BuildTool(randomId(), BuildTool.Type.Maven, runtime.getMavenVersion());
         this.exclusions = exclusions;
         sizeThresholdMb = thresholdMb;
         this.mavenSession = session;
         this.settingsDecrypter = settingsDecrypter;
+    }
+
+    public List<SourceFile> listSourceFiles(MavenProject mavenProject, Iterable<NamedStyles> styles,
```

Comment:
this method is starting to feel uncomfortably long, like one of those things that you are afraid to touch

Reply:
Yeah, I am going to clean this up and extract logic into easier-to-reason-about methods.